docs/source/design.rst

```diff
@@ -1,7 +1,151 @@
-.. _internals:
+.. _design:
 
-*********************
-Ibis design internals
-*********************
+Design
+======
 
-More to come here.
+Primary Goals
+*************
+
+#. Type safety
+#. Expressiveness
+#. Composability
+#. Familiarty
+
+Flow of execution
+-----------------
+
+#. User writes expression
+#. Each method or function call builds a new expression
+#. Expressions are type checked as you create them
+#. Some expressions have some optimizations that happen as the user builds them
+   (EXAMPLE)
+#. Some optimizations happen at compile time? (EXAMPLE)
+#. Expressions are compiled
+#. The SQL string that generated by the compiler is sent to the database and
+   executed
+#. The database returns some data that is then turned into a pandas DataFrame
+   by ibis
+
+
+Expressions
+-----------
+
+The main user-facing component of ibis is expressions. The base class of all
+expressions in ibis is the :class:`~ibis.expr.types.Expr` class.
+
+Expressions provide the user facing API, defined in ``ibis/expr/api.py``
```

Comment:
you could actually put in the links to code

Reply:
yep, will do